File: frontend/src/types/adminUser.ts
import type { VocabItem } from './vocab'

export interface AdminUserItem {
  id: number
  username: string
  email: string | null
  status: 'active' | 'disabled'
  created_at: string
  last_login_at: string | null
  vocab_count: number
  query_count: number
}

export interface AdminUserListResponse {
  items: AdminUserItem[]
  total: number
  page: number
  page_size: number
}

export interface AdminUserCreateResponse {
  user: AdminUserItem
  temporary_password: string
}

export interface QueryLogEntry {
  word: string
  status: string | null
  dictionary_id: number | null
  duration_ms: number | null
  created_at: string
}

export interface AdminUserDetail {
  user: AdminUserItem
  vocab_items: VocabItem[]
  recent_queries: QueryLogEntry[]
}

File: backend/app/schemas/admin_user.py
from datetime import datetime

from pydantic import BaseModel, EmailStr, Field

from app.schemas.vocab import VocabItemOut


class AdminUserCreateRequest(BaseModel):
    username: str = Field(min_length=3, max_length=64)
    email: EmailStr | None = None


class AdminUserOut(BaseModel):
    id: int
    username: str
    email: str | None
    status: str
    created_at: datetime
    last_login_at: datetime | None
    vocab_count: int
    query_count: int


class AdminUserCreateResponse(BaseModel):
    user: AdminUserOut
    temporary_password: str


class AdminUserListResponse(BaseModel):
    items: list[AdminUserOut]
    total: int
    page: int
    page_size: int


class ResetPasswordResponse(BaseModel):
    temporary_password: str


class QueryLogOut(BaseModel):
    word: str
    status: str | None
    dictionary_id: int | None
    duration_ms: int | None
    created_at: datetime

    model_config = {"from_attributes": True}


class AdminUserDetailResponse(BaseModel):
    user: AdminUserOut
    vocab_items: list[VocabItemOut]
    recent_queries: list[QueryLogOut]
